frontend/src/pages/PipelineDetails.tsx

```diff
@@ -236,48 +242,91 @@ class PipelineDetails extends Page<{}, PipelineDetailsState> {
 
   public async load(): Promise<void> {
     this.clearBanner();
-    const pipelineId = this.props.match.params[RouteParams.pipelineId];
+    const fromRunId = new URLParser(this.props).get(QUERY_PARAMS.fromRunId);
 
-    let pipeline: ApiPipeline;
-    let templateResponse: ApiGetTemplateResponse;
+    let pipeline: ApiPipeline | null = null;
+    let templateYaml = '';
+    let template: Workflow | undefined;
+    let breadcrumbs: Array<{ displayName: string, href: string }> = [];
+    const toolbarActions = [...this.props.toolbarProps.actions];
+    let pageTitle = '';
 
-    try {
-      pipeline = await Apis.pipelineServiceApi.getPipeline(pipelineId);
-    } catch (err) {
-      await this.showPageError('Cannot retrieve pipeline details.', err);
-      logger.error('Cannot retrieve pipeline details.', err);
-      return;
-    }
+    // If fromRunId is specified, load the run and get the pipeline template from it
+    if (fromRunId) {
+      try {
+        const runDetails = await Apis.runServiceApi.getRun(fromRunId);
+        templateYaml = RunUtils.getPipelineSpec(runDetails.run) || '';
 
-    try {
-      templateResponse = await Apis.pipelineServiceApi.getTemplate(pipelineId);
-    } catch (err) {
-      await this.showPageError('Cannot retrieve pipeline template.', err);
-      logger.error('Cannot retrieve pipeline details.', err);
-      return;
+        const relatedExperimentId = RunUtils.getFirstExperimentReferenceId(runDetails.run);
+        let experiment: ApiExperiment | undefined;
+        if (relatedExperimentId) {
+          experiment = await Apis.experimentServiceApi.getExperiment(relatedExperimentId);
+        }
+
+        // Build the breadcrumbs, by adding experiment and run names
+        if (experiment) {
+          breadcrumbs.push(
+            { displayName: 'Experiments', href: RoutePage.EXPERIMENTS },
+            {
+              displayName: experiment.name!,
+              href: RoutePage.EXPERIMENT_DETAILS.replace(':' + RouteParams.experimentId, experiment.id!)
+            });
+        } else {
+          breadcrumbs.push(
+            { displayName: 'All runs', href: RoutePage.RUNS }
+          );
+        }
+        breadcrumbs.push({
+          displayName: runDetails.run!.name!,
+          href: RoutePage.RUN_DETAILS.replace(':' + RouteParams.runId, fromRunId),
+        });
+        pageTitle = 'Pipeline details';
```

Comment:
nit: maybe `Embedded Pipeline details`
or
`Embedded Pipeline` ?

The word doesn't have to be "embedded" either

Reply:
I don't know if that's easily understood by the user. We can change this later if we have better ideas, but for now I think [the spec](https://github.com/kubeflow/pipelines/issues/75#issuecomment-440441381) is fine.